### ISSUE TITLE
generator-star-spacing support for methods in classes

### ISSUE THREAD

**Issue by julien-f** [eslint/eslint#2351]
The following configuration works well for both function and methods in object literals but fails for methods in classes:

```json
{
  "env": {
    "es6": true
  },

  "rules": {
    "generator-star-spacing": [2, "after"]
  }
}
```

Errors are not detected for methods in classes:

```javascript
class className {
  *methodName() {}
}
```

But they are for functions and methods in object literals:

```javascript
function*funcName() {}

({
  *methodName() {}
})
```

**Comment by btmills** [comment]
Working on this.
